examples/tv-app/tv-common/clusters/messages/MessagesManager.h

```diff
@@ -21,6 +21,97 @@
+ *
+ */
+
+struct CachedMessageOption
+{
+    uint32_t mId;
+    std::string mLabel;
+    chip::app::Clusters::Messages::MessageResponseOption mOption;
+
+    CachedMessageOption(uint32_t id, std::string label) :
+        mId(id), mLabel(label),
+        mOption{ chip::Optional<uint32_t>(mId), chip::Optional<chip::CharSpan>(chip::CharSpan::fromCharString(mLabel.c_str())) }
+    {
+        // ChipLogProgress(Controller, "CachedMessageOption constructor id=%d label=%s", mId, mLabel.c_str());
+    }
+
+    chip::app::Clusters::Messages::MessageResponseOption GetMessageOption() { return mOption; }
+
+    ~CachedMessageOption()
+    {
+        // ChipLogProgress(Controller, "CachedMessageOption destructor");
+    }
+};
+
+struct CachedMessage
+{
+    const chip::app::Clusters::Messages::MessagePriorityEnum mPriority;
+    const chip::BitMask<chip::app::Clusters::Messages::MessageControlBitmap> mMessageControl;
+    const chip::app::DataModel::Nullable<uint32_t> mStartTime;
+    const chip::app::DataModel::Nullable<uint16_t> mDuration;
+
+    std::string mMessageText;
+
+    uint8_t * messageIdBuffer = nullptr;
+    chip::ByteSpan mMessageId;
+
+    CachedMessage(const chip::ByteSpan & messageId, const chip::app::Clusters::Messages::MessagePriorityEnum & priority,
+                  const chip::BitMask<chip::app::Clusters::Messages::MessageControlBitmap> & messageControl,
+                  const chip::app::DataModel::Nullable<uint32_t> & startTime,
+                  const chip::app::DataModel::Nullable<uint16_t> & duration, std::string messageText) :
+        mPriority(priority),
+        mMessageControl(messageControl), mStartTime(startTime), mDuration(duration), mMessageText(messageText)
+    {
+        messageIdBuffer = new uint8_t[messageId.size()];
+        VerifyOrReturn(messageIdBuffer != nullptr, ChipLogProgress(Controller, "CachedMessage messageIdBuffer alloc failed"));
+        memcpy(messageIdBuffer, messageId.data(), messageId.size());
+        mMessageId = chip::ByteSpan(messageIdBuffer, messageId.size());
+    }
+
+    void AddOption(CachedMessageOption * option)
+    {
+        mOptions.push_back(*option);
+        mResponseOptions.push_back(option->GetMessageOption());
+    }
+
+    chip::app::Clusters::Messages::Structs::MessageStruct::Type GetMessage()
+    {
+        chip::app::DataModel::List<chip::app::Clusters::Messages::MessageResponseOption> options(mResponseOptions.data(),
+                                                                                                 mResponseOptions.size());
+        chip::app::Clusters::Messages::Structs::MessageStruct::Type message{
+            mMessageId,
+            mPriority,
+            mMessageControl,
+            mStartTime,
+            mDuration,
+            chip::CharSpan::fromCharString(mMessageText.c_str()),
+            chip::Optional<chip::app::DataModel::List<chip::app::Clusters::Messages::MessageResponseOption>>(options)
```

Comment:
So this will return empty list if input did not have options at all.  Is that expected per spec?

Reply:
removed empty list